### PR TITLE
Ofir/sat 1478

### PR DESCRIPTION
**Pull request by OfirZinkerz**
implement my profile page

### CODE REVIEW

File: src/components/myProfile/components/changePassword/changePassword.component.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    if (!authform.$invalid) {
+                        AuthService.changePassword(vm.changePasswordData).then(function () {
+                            $timeout(function (res) {
+                                console.log('res ', res);
+                                type = 'success';
+                                msg = 'MY_PROFILE.PASSWORD_SAVE_SUCCESS';
+                                showToast(type, msg);
+                            }, 10);
+                        }, function (err) {
+                            console.log('err: ', err);
```
Comment (orenor10) [inline src/components/myProfile/components/changePassword/changePassword.component.js]: use $log if you need logging
Please remove console.log

File: src/components/myProfile/components/changePassword/changePassword.component.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+                    if (vm.changePasswordData.newPassword !== vm.changePasswordData.newPasswordConfirm) {
+                        vm.changePasswordData.newPasswordConfirm = undefined;
+                        return;
+                    }
+
+                    if (!authform.$invalid) {
+                        AuthService.changePassword(vm.changePasswordData).then(function () {
+                            $timeout(function (res) {
+                                console.log('res ', res);
```
Comment (orenor10) [inline src/components/myProfile/components/changePassword/changePassword.component.js]: use $log if you need logging
Please remove console.log

File: src/components/myProfile/components/updateProfile/updateProfile.component.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                vm.updateProfile = function (profileform) {
+                    var type, msg;
+
+                    if (profileform.$valid && profileform.$dirty) {
+                        UserProfileService.setProfile(vm.profileData).then(function () {
+                            $timeout(function () {
+                                type = 'success';
+                                msg = 'MY_PROFILE.PROFILE_SAVE_SUCCESS';
+                                showToast(type, msg);
+                            }, 10);
```
Comment (orenor10) [inline src/components/myProfile/components/updateProfile/updateProfile.component.js]: why timeout for 10 sec?? do you need waiting time or just digest?

File: src/components/myProfile/components/updateProfile/updateProfile.component.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                            $timeout(function () {
+                                type = 'error';
+                                if (err.code === 'NETWORK_ERROR') {
+                                    msg = 'MY_PROFILE.NO_INTERNET_CONNECTION_ERR';
+                                    showToast(type, msg);
+                                } else {
+                                    msg = 'MY_PROFILE.ERROR_OCCURRED';
+                                    showToast(type, msg);
+                                }
+                            }, 10);
```
Comment (orenor10) [inline src/components/myProfile/components/updateProfile/updateProfile.component.js]: 10 sec....??

File: src/components/myProfile/controllers/toast.controller.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+(function (angular) {
```
Comment (orenor10) [inline src/components/myProfile/controllers/toast.controller.js]: toast controller should not be a part of user profile

File: src/components/myProfile/module.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        'znk.infra.svgIcon',
+        'znk.infra.general',
+        'znk.infra.storage',
+        'znk.infra.user'
+    ])
+    .config([
+        'SvgIconSrvProvider',
+        function (SvgIconSrvProvider) {
+            var svgMap = {
+                'myProfile-icon': 'components/myProfile/svg/profile-icon.svg',
```
Comment (orenor10) [inline src/components/myProfile/module.js]: Please rename the svg file name 
add 'myProfile-' prefix to svg file names

File: src/components/myProfile/scss/_myProfile.scss [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+                    top: -25px;
+                    background-color: $bgColor32;
+                    color: $fontColor1;
+                    font-size: 13px;
+                    line-height: 24px;
+                    &:after {
+                        content: '';
+                        position: absolute;
+                        border: {
+                            color: transparent;
```
Comment (orenor10) [inline src/components/myProfile/scss/_myProfile.scss]: use $transparent

File: src/components/myProfile/scss/_myProfile.scss [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+            display: flex;
+            justify-content: flex-end;
+            margin-right: 50px;
+            outline: none;
+            .save-pass-btn {
+                width: 15%;
+                margin: 0;
+                box-shadow: none;
+                height: 30px;
+                background: transparent;
```
Comment (orenor10) [inline src/components/myProfile/scss/_myProfile.scss]: use $transparent

File: src/components/myProfile/scss/_toast.scss [hunk 9]
```diff
@@ -0,0 +1,68 @@
+md-toast.toast-wrap{
+    top: 100px !important;
+    right: calc(50% - 600px) !important;
```
Comment (orenor10) [inline src/components/myProfile/scss/_toast.scss]: why !important???

File: src/components/myProfile/scss/_toast.scss [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    width: 400px;
+    height: 60px;
+    display: flex;
+    background-color: #FFFFFF;
```
Comment (orenor10) [inline src/components/myProfile/scss/_toast.scss]: user color var

File: src/components/myProfile/scss/_toast.scss [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+    border: 2px solid #87CA4D;
```
Comment (orenor10) [inline src/components/myProfile/scss/_toast.scss]: user color var

File: src/components/myProfile/scss/_toast.scss [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    border-radius: 10px;
+
+    .icon-wrap{
+        width: 60px;
+        margin: -8px;
+        padding: 8px;
+        background: #87CA4D;
```
Comment (orenor10) [inline src/components/myProfile/scss/_toast.scss]: user color var

File: src/components/myProfile/scss/_toast.scss [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        svg-icon{
+            width: 40px;
+            height: 40px;
+            .complete-v-icon-svg{
+                .st1, .st3 {
+                    fill: #87CA4D;
```
Comment (orenor10) [inline src/components/myProfile/scss/_toast.scss]: use color var
not good. you should not format st1 & st3

File: src/components/myProfile/scss/_toast.scss [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+                }
+            }
+        }
+    }
+
+    .md-toast-content{
+        background: #FFFFFF;
```
Comment (orenor10) [inline src/components/myProfile/scss/_toast.scss]: color var

File: src/components/myProfile/scss/_toast.scss [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+        box-shadow: none;
+        color: #4F4D4D;
```
Comment (orenor10) [inline src/components/myProfile/scss/_toast.scss]: color var

File: src/components/myProfile/scss/_toast.scss [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+    .close-toast-wrap{
+        position: absolute;
+        right: 3px;
+        top: -15px;
+        width: 20px;
+        .close-pop-svg{
+            width: 10px;
+            .st1 {
+                stroke: #4F4D4D;
```
Comment (orenor10) [inline src/components/myProfile/scss/_toast.scss]: color var

File: src/components/znkHeader/components/znkHeader/_znkHeader.scss [hunk 17]
```diff
@@ -49,16 +47,16 @@
...
 
                 .pending-purchase-icon-wrapper {
                     width: 16px;
                     height: 16px;
                     margin-right: 10px;
-                    float: left;
+                    float:left;
                     & + span {
-                        @include text-style(16px, null, Lato, Light, null, null, italic);
+                        @include text-style(16px, null, Lato ,Light, null, null, italic);
```
Comment (orenor10) [inline src/components/znkHeader/components/znkHeader/_znkHeader.scss]: is 'Light' (for font-weight) is declared?

File: src/components/myProfile/components/changePassword/changePassword.component.js (hunk 1 above)
Comment (OfirZinkerz) [inline src/components/myProfile/components/changePassword/changePassword.component.js]: deleted it

File: src/components/invitation/directives/_invitation-manager.scss [hunk 18]
```diff
@@ -32,9 +32,11 @@ invitation-manager {
     }
 
     .teacher-icon-wrap {
-        background-color: $bgColor33;
+        border: 1px solid $borderColor24;
+        background-color: transparent;
```
Comment (OfirZinkerz) [inline src/components/invitation/directives/_invitation-manager.scss]: done

File: src/components/myProfile/components/changePassword/changePassword.component.js (hunk 2 above)
Comment (OfirZinkerz) [inline src/components/myProfile/components/changePassword/changePassword.component.js]: deleted

File: src/components/myProfile/components/updateProfile/updateProfile.component.js (hunk 3 above)
Comment (OfirZinkerz) [inline src/components/myProfile/components/updateProfile/updateProfile.component.js]: remove it

File: src/components/myProfile/components/updateProfile/updateProfile.component.js (hunk 4 above)
Comment (OfirZinkerz) [inline src/components/myProfile/components/updateProfile/updateProfile.component.js]: removed

File: src/components/myProfile/controllers/toast.controller.js (hunk 5 above)
Comment (OfirZinkerz) [inline src/components/myProfile/controllers/toast.controller.js]: moved to different component

File: src/components/myProfile/module.js (hunk 6 above)
Comment (OfirZinkerz) [inline src/components/myProfile/module.js]: renamed

File: src/components/myProfile/scss/_myProfile.scss [hunk 19]
```diff
@@ -0,0 +1,257 @@
...
+        min-height: 40px;
+        border-top-left-radius: 20px;
+        border-top-right-radius: 20px;
+
+        &:not(.md-menu-toolbar){
+            background-color: $bgColor5;
+        }
+
+        .close-popup-wrap {
+            @include absolute-position(-35px, -35px);
```
Comment (OfirZinkerz) [inline src/components/myProfile/scss/_myProfile.scss]: yep, and it's responsive

File: src/components/myProfile/scss/_myProfile.scss (hunk 7 above)
Comment (OfirZinkerz) [inline src/components/myProfile/scss/_myProfile.scss]: ok

File: src/components/myProfile/scss/_myProfile.scss (hunk 8 above)
Comment (OfirZinkerz) [inline src/components/myProfile/scss/_myProfile.scss]: done

**Pull request merged**